packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -0,0 +1,210 @@
+    // zoom only allows retrieval by meeting so we need meeting ID and file ID to retrieve
+    const meetingId = id
+    const fileId = query.recordingId
+    const GET_MEETING_FILES = `/meetings/${meetingId}/recordings`
+
+    const downloadUrlPromise = new Promise((resolve) => {
+      this.client
+        .get(`${BASE_URL}${GET_MEETING_FILES}`)
+        .auth(token)
+        .request((err, resp) => {
+          if (err) {
+            this._downloadError(resp, done)
+            return
+          }
+          const file = resp.body.recording_files.find(file => file.id === fileId)
+          if (!file || !file.download_url) {
+            return this._downloadError(resp, done)
+          }
+          resolve(file.download_url)
+        })
+    })
+    downloadUrlPromise.then((downloadUrl) => {
+      this.client
+        .get(`${downloadUrl}?access_token=${token}`)
+        .request()
+        .on('response', (resp) => {
+          if (resp.statusCode !== 200) {
+            done(this._error(null, resp))
+          } else {
+            resp.on('data', (chunk) => done(null, chunk))
+          }
+        })
+        .on('end', () => done(null, null))
+        .on('error', (err) => {
+          logger.error(err, 'provider.zoom.download.error')
+          done(err)
+        })
+    })
+  }
+
+  size ({ id, token, query }, done) {
+    const meetingId = id
+    const fileId = query.recordingId
+    const GET_MEETING_FILES = `/meetings/${meetingId}/recordings`
+
+    return this.client
+      .get(`${BASE_URL}${GET_MEETING_FILES}`)
+      .auth(token)
+      .request((err, resp, body) => {
+        if (err || resp.statusCode !== 200) {
+          return this._downloadError(resp, done)
+        }
+        const file = resp.body.recording_files.find(file => file.id === fileId)
+        if (!file) {
+          return this._downloadError(resp, done)
+        }
+        done(null, file.file_size)
+      })
+  }
+
+  _adaptData (results, dateRange) {
+    if (!results || results.length === 0) {
+      return { items: [] }
+    }
+
+    const data = {
+      // is this nextPagePath ever used? It looks like pagination just uses the querystring / cursor key
+      nextPagePath: adapter.getQuery(results, dateRange),
+      items: []
+    }
+
+    results.forEach(res => {
+      res.meetings.forEach(meeting => {
+        meeting.recording_files.forEach((record, index) => {
+          data.items.push({
+            name: adapter.getItemName(meeting, record, index),
+            mimeType: adapter.getMimeType(record),
+            id: adapter.getId(record),
+            requestPath: adapter.getRequestPath(record),
+            modifiedDate: adapter.getStartDate(record),
+            size: adapter.getSize(record),
+            custom: adapter.getCustomFields(record)
+          })
+        })
+      })
+    })
+
+    return data
+  }
+
+  _error (err, response) {
+    // token expiry error doesn't seem to be returned correctly
```

Comment:
i'm not 100% sure why the token expiry isn't being recognized correctly, any thoughts / feedback would be helpful. thanks!